Templates/Templates/AutoDecodable.stencil

```diff
@@ -0,0 +1,96 @@
+import Foundation
+
+extension KeyedDecodingContainer {
+  public func decodeOrDefault<T: Decodable>(_ type: T.Type, forKey key: Key, defaultValue: T) throws -> T {
+    guard let value: T = try decodeIfPresent(type, forKey: key) else {
+      return defaultValue
+    }
+    return value
+  }
+}
+
+{# these are a giant line because of spacing, to debug just seperate the if into lines#}
+{% macro defaultValueFor variable %}{% set typename %}{{ variable.typeName.name }}{% endset %}{% if variable|annotated:"codingDefaultValue" %}{{ variable.annotations.codingDefaultValue }}){% elif variable.isOptional %}nil){% elif variable.isArray %}[]){% elif typename == "String" %}""){% elif typename == "Int" or typename == "Double" or typename == "Float" or typename == "NSNumber" %}0){% elif typename == "Bool" %}false){% endif %}{% endmacro %}
+{% macro decodeNameWithTransform name transform %}{% if transform == "camelToSnake" or not transform %}{{ name|camelToSnakeCase }}{% elif transform == "camelToSnake:idToUUID" %}{{ name|camelToSnakeCase|replace:"id","uuid" }}{% elif transform == "idToUUID" %}{{ name|replace:"id","uuid" }}{% endif %}{% endmacro %}
+{% macro keyedNameFor type  variable %}{% if variable|annotated:"codingKey" %} = "{{ variable.annotations.codingKey }}"{% elif type|annotated:"codingKeyTransform" %}= "{% call decodeNameWithTransform variable.name type.annotations.codingKeyTransform %}"{% endif %}{% endmacro %}
+
+{# the keys to use to decoding #}
+{% macro codingKeys type %}
+  private enum CodableKeys: String, CodingKey {
```

Comment:
I found it useful to have `DecodingKeys` and `EncodingKeys` and make them public. This way they can be used elsewhere in the code.